### ISSUE TITLE
Rares in  instances are not working properly

### ISSUE THREAD

**Issue by T1ti** [azerothcore/azerothcore-wotlk#4621]
##### CURRENT BEHAVIOUR:

In dungeons, rares are either spawned all the time, or never if they are missing.
This mostly affects vanilla instances as blizzard didn't really add any rares to instances in later expensions, one known exception is karazhan.

##### EXPECTED BLIZZLIKE BEHAVIOUR:

Rares don't always spawn, they have a % chance to be  spawned in the instance.

##### STEPS TO REPRODUCE THE PROBLEM:
Check any rare creature in a dungeon
Some examples :
- Skul in stratholme main entrance always spawns
-Jed in UBRS

##### EXTRA NOTES:

https://github.com/TrinityCore/TrinityCore/issues/24437 Issue from TC, still open and unsolved.

##### AC HASH/COMMIT:
a4115862de1cd6f37b70969c87df79f327457dd1

**Comment by T1ti** [comment]
Update : 
I found that many rares have an SAI that give them on spawn a 70% chance to despawn, which tries to recreate the missing system, but badly.

**Comment by Zoidwaffle** [comment]
I'm adding a reference to your own work here: https://github.com/azerothcore/azerothcore-wotlk/pull/4639 (rares in Vanilla).

I can understand that the "SAI despawn" might be a sort of hack, but I don't think it's an ugly hack - just a little work-around.

So I'm not sure this issue should remain open - unless perhaps changing it to where the rares are still missing? After your PR above, then it's "only" TBC and WotLK dungeons missing.

**Comment by T1ti** [comment]
Another option is to give them a chance to spawn with pool_creature, instead of giving them a chance to despawn from SAI.
That's how vmangos handles it, there are some rares that share spawn points with normal creatures and this feature is required in this case

**Comment by T1ti** [comment]
Second point : We currently can't have multiple spawn points using the SAI despawn method
This is how vmangos handles them : works fine with multiple spawn point and shared spawns
![image](https://user-images.githubusercontent.com/40864460/109408546-0cf91e80-798b-11eb-8b21-472c59ddf624.png)

**Comment by Zoidwaffle** [comment]
I agree, in some cases the SAI fix could be less optimal. But having rares with SAI > not having rares at all :heart:

If the way vmangos does it can be ported to AC, it could be even better.
